### PR TITLE
Correctly emit prologue directives in constructors

### PR DESCRIPTION
**Pull request by ShyykoSerhiy**
## Fixed "Directive prologues are not preserved"
Fixes issue #3454

### CODE REVIEW

File: src/compiler/emitter.ts [hunk 1]
```diff
@@ -4219,6 +4219,9 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, Promi
                 scopeEmitStart(node, "constructor");
                 increaseIndent();
                 if (ctor) {
+                    // Emit all the directive prologues (like "use strict").  These have to come before
+                    // any other preamble code we write (like parameter initializers).
+                    var startIndex = emitDirectivePrologues(ctor.body.statements, /*startWithNewLine*/ true);
```
Comment (mhegazy) [inline src/compiler/emitter.ts]: Please define startIndex outside the if statement, and initialize it appropriately.

Comment (mhegazy) [inline src/compiler/emitter.ts]: also use let and const instead of var.

**Comment by DanielRosenwasser** [comment]
By the way @ShyykoSerhiy, when I suggested you format your issues in the way that I did, I meant for the big text to go into the title area.

**Comment by mhegazy** [comment]
took care of the comments locally. thanks @ShyykoSerhiy for the PR.

**Pull request merged**